email_mailbox/src/utils/electronEventInterface.js

```diff
@@ -154,33 +175,37 @@ const parseAndStoreEventsBatch = async ({ events, hasMoreEvents }) => {
         sendFetchEmailsErrorMessage();
       }
     }
+    i++;
   }
 
   const rowIdsFiltered = rowIds.filter(rowId => !!rowId);
-  if (rowIdsFiltered.length || (events.length && !rowIdsFiltered.length)) {
+  if (rowIdsFiltered.length || (eventsSize && !rowIdsFiltered.length)) {
```

Comment:
To avoid nested code, you could use 
```
if (!rowIdsFiltered.length && !(eventsSize && !rowIdsFiltered.length)) {
  continue;
}
if (rowIdsFiltered.length) await setEventAsHandled(rowIdsFiltered);
if (shouldBreak) {
  break;
}
```
this way you avoid 2 nested levels and you break the instead of checking again in the while condition